### ISSUE TITLE
More powerful schema kind

### ISSUE THREAD

**Issue by echatav** [morphismtech/squeal#4]
At the moment a schema is kinded as a `TablesType`, a list of named tables.
```
type TablesType = [(Symbol,[(Symbol,ColumnType)])]
```
This is wrong in two ways:
1) `Schema`s contain more than just tables, they also contain views, user-defined functions, operators and types.
2) A database can have more than 1 schema.
From the [PostgreSQL docs](https://www.postgresql.org/docs/9.1/static/ddl-schemas.html):
> A database contains one or more named schemas, which in turn contain tables. Schemas also contain other kinds of named objects, including data types, functions, and operators. The same object name can be used in different schemas without conflict; for example, both schema1 and myschema can contain tables named mytable. Unlike databases, schemas are not rigidly separated: a user can access objects in any of the schemas in the database he is connected to, if he has privileges to do so.

In principle this could be a record datakind like
```haskell
data SchemaType = SchemaType
  { tables :: TablesType
  , views :: TablesType
  , functions :: [(Symbol,[ColumnType],[ColumnType])]
  , typedefs :: [(Symbol,ColumnType)]
  }
type Database = [(Symbol,SchemaType)]
```
Unfortunately we don't get free type families from the record accessors as far as I know so you'd also have to define e.g.
```haskell
GetTables :: SchemaType -> TablesType
ModifyTables :: (TablesType -> TablesType) -> SchemaType -> SchemaType
```
It might be easier to bite off small pieces of that, like support multiple schemas separately from having more than tables in schemas, and in doing the latter, take on each of `views`, `functions`, and `typedefs` separately.

**Comment by echatav** [comment]
sketch of support for composite and enum types and `VIEW`s.

```haskell
data ObjectType
  = Table TableType
  | View RelationType
  | Enum [Symbol]
  | Composite [(Symbol, PGType)]

type SchemaType = [(Symbol, ObjectType)]

createTypeEnum
  :: Aliased (NP Alias) (ty ::: labels)
  -> Definition schema (Create ty (Enum labels) schema)

createTypeComposite
  :: AllNull fields
  => Aliased (NP (Aliased TypeExpression)) (ty ::: fields)
  -> Definition schema (Create ty (Composite (BaseTypesOf fields)) schema)

createView
  :: Aliased (Query schema params) (view ::: relation)
  -> Definition schema (Create view (View relation) schema)

table
  :: Has table (TablesOf schema) tablety
  => Alias table
  -> FromClause schema params '[table ::: TableToRelation tablety]

view
  :: Has view (ViewsOf schema) relation
  => Alias view
  -> FromClause schema params '[view ::: relation]

composite
  :: Has ty (CompositesOf schema) fields
  => Alias ty
  -> TypeExpression schema (UnsafePGType ty)

enum
  :: Has ty (EnumsOf schema) labels
  => Alias ty
  -> TypeExpression schema (UnsafePGType ty)
```

**Comment by echatav** [comment]
Partly solved by #33
